test/integration/singlecluster/controller/dra/suite_test.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dra
+
+import (
+	"context"
+	"testing"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	resourcev1beta2 "k8s.io/api/resource/v1beta2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	config "sigs.k8s.io/kueue/apis/config/v1beta1"
+	qcache "sigs.k8s.io/kueue/pkg/cache/queue"
+	schdcache "sigs.k8s.io/kueue/pkg/cache/scheduler"
+	"sigs.k8s.io/kueue/pkg/controller/core"
+	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
+	"sigs.k8s.io/kueue/pkg/dra"
+	"sigs.k8s.io/kueue/pkg/features"
+	"sigs.k8s.io/kueue/pkg/scheduler"
+	"sigs.k8s.io/kueue/pkg/webhooks"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client
+	ctx       context.Context
+	fwk       *framework.Framework
+)
+
+func TestAPIs(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	ginkgo.RunSpecs(t,
+		"DRA Controller Suite",
+	)
+}
+
+var _ = ginkgo.BeforeSuite(func() {
+	features.SetFeatureGateDuringTest(ginkgo.GinkgoTB(), features.DynamicResourceAllocation, true)
+
+	fwk = &framework.Framework{
+		WebhookPath: util.WebhookPath,
+		APIServerFeatureGates: []string{
+			"DynamicResourceAllocation=true",
+		},
+		APIServerRuntimeConfig: []string{
+			"resource.k8s.io/v1beta2=true",
```

Comment:
Folks, can we please start with supporting v1 only? My thinking:
1. we already have k8s v1.34 with v1 released, so the path for "early adopters" already exist
2. v1beta2 is disabled by default in the core k8s as of 1.33 so users on most clouds (like GKE where the control-plane manifest is locked) would not benefit from this, just on-prem. The on prem users are often early adopters who can afford using v1.34
3. we will need to migrate to v1 eventually, which just delays the process, and may slow down the integration in Kueue if we need some new fields
4. I have not yet had an explicit user request to support DRA with Kueue, which suggests the early adopters are probably ok to wait a little bit

cc @alaypatel07 @klueska @tenzen-y @kannon92 @mwysokin

Reply:
I am fine with supporting 1.34 and on for this feature.

This may mess up our k8s support though.

I think we are saying that DRA + Kueue would only be available on 1.34 and on (so this feature would not be valid for 1.31, 1.32 or 1.33).

Reply:
> I think we are saying that DRA + Kueue would only be available on 1.34 and on (so this feature would not be valid for 1.31, 1.32 or 1.33).

Yes, we can document that. It would not be valid for 1.31 anyway as 1.31 is v1beta1.

For 1.32 and 1.33, yes, but note that while DRA is Beta it remains disabled, which makes it anyway unsable on most clouds where the feature gates are locked: https://github.com/kubernetes/kubernetes/blob/69aca29e6def5873779cbd392bd9a1bad124c586/pkg/features/kube_features.go#L1236-L1240

Reply:
GKE allows it but the user has to explicitly enable it. EKS I believe may enable it by default in order to support the NVIDIA GB200 on 1.33.

All new endpoints in K8s are disabled at beta by default. *Feature gates* are enabled by default at beta. That means that beta fields in existing `v1` APIs will become available. But `v1betaX` endpoints are disabled by default, because the encoding in the API object meta data means massive migrations for customers when they write manifests that way.

I think it's OK to do v1 only, unless we have specific use cases like GB200 that people need to do in 1.33.

Reply:
I can confirm EKS added it to support GB200 on 1.33